lib/pc_patch.c

```diff
@@ -608,3 +608,117 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+
+/**
+* Function used by pc_patch_set_schema. Copy the data of p1 in p2. Assume that
+* p1 and p2 schemas are compatible, meaning that the dimensions in p2 that are
+* also in p1 have the same interpretations. The value "defaults" is used for
+* dimensions that are not in p1.
+*/
+static void
+pc_patch_point_copy(const PCPOINT *p1, PCDIMENSION **dims,
+		double defaults, PCPOINT *p2)
+{
+	size_t i;
+	for ( i = 0; i < p2->schema->ndims; i++ )
+	{
+		PCDIMENSION *d1 = dims[i];
+		PCDIMENSION *d2 = p2->schema->dims[i];
+
+		if ( d1 )
+		{
+			uint8_t *data2 = p2->data + d2->byteoffset;
+			uint8_t *data1 = p1->data + d1->byteoffset;
+			memcpy(data2, data1, d1->size);
+		}
+		else
+		{
+			pc_point_set_double(p2, d2, defaults);
+		}
+	}
+}
+
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_schema(PCPATCH *patch, const PCSCHEMA *new_schema, double defaults)
+{
+	PCDIMENSION** new_dimensions = new_schema->dims;
+	PCDIMENSION* old_dimensions[new_schema->ndims];
+	const PCSCHEMA *old_schema = patch->schema;
+	PCPATCH_UNCOMPRESSED *paout;
+	PCPOINT opt, npt;
+	PCPATCH *pain;
+	size_t i, j;
+
+	for ( j = 0; j < new_schema->ndims; j++ )
+	{
+		PCDIMENSION *ndim = new_dimensions[j];
+		PCDIMENSION *odim = pc_schema_get_dimension_by_name(
+				old_schema, ndim->name);
+		if ( odim && ndim->interpretation != odim->interpretation )
+		{
+			pcerror("dimension interpretations are not matching");
+			return NULL;
+		}
+		old_dimensions[j] = odim;
+	}
```

Comment:
a default data buffer or a default PCPoint could be built here, so as to prevent the reinterpretation of the default values whenever it is needed.